QuizApp/QuizSession/Views/QuestionView.swift

```diff
@@ -0,0 +1,125 @@
+import Combine
+import UIKit
+import SnapKit
+
+class QuestionView: UIView {
+
+    private let margins = 10
+    private let questionInsets = 20
+    private let buttonTopOffset = 30
+
+    private var cancellables = Set<AnyCancellable>()
+    private var answers: [AnswerModel]!
+    private var correctAnswerId: Int!
+    private var questionLabel: UILabel!
+    private var stackView: UIStackView!
+
+    @Published private var isCorrectAnswer: Bool = false
+
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+
+        buildViews()
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    func setQuestion(question: QuestionModel) {
+        self.answers = question.answers
+        self.correctAnswerId = question.correctAnswerId
+        questionLabel.text = question.question
+        setAnswerButtons()
+    }
+
+    func setAnswerButtons() {
+        stackView.subviews.forEach { $0.removeFromSuperview() }
+
+        answers.forEach { answer in
+            let answerButton = Button(id: answer.id)
+            let title = NSAttributedString(
+                string: answer.answer,
+                attributes: [.font: UIFont.systemFont(ofSize: 20, weight: .semibold),
+                             .foregroundColor: UIColor.white]
+            )
+            answerButton.setAttributedTitle(title, for: .normal)
+            answerButton.contentHorizontalAlignment = UIControl.ContentHorizontalAlignment.leading
+            answerButton.titleEdgeInsets = UIEdgeInsets(top: 0, left: 25, bottom: 0, right: 0)
+            answerButton.layer.cornerRadius = 30
+            answerButton.backgroundColor = .white.withAlphaComponent(0.3)
+
+            answerButton
+                .tap
+                .sink { [weak self] _ in
+                    guard let self = self else { return }
+
+                    let isCorrect = answer.id == self.correctAnswerId
+                    answerButton.backgroundColor = isCorrect ? .green : .red
+
+                    if !isCorrect {
+                        self.colorCorrectAnswer()
+                    }
+
```

Comment:
Why is this needed when you have `answerButton.backgroundColor = isCorrect ? .green : .red` ?

Reply:
In case the chosen answer is not correct, with ```answerButton.backgroundColor = isCorrect ? .green : .red``` it's color changes to red, but then I have to go through all the answers and color the correct one green

Reply:
That logic is a bit confusing, and not obvious. 
Instead of having both `answerButton.backgroundColor = isCorrect ? .green : .red ` and `colorCorrectAnswer()`, keep only the function but then go through all buttons and color them in the appropriate color.